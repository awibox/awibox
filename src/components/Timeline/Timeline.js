import React from 'react';
import PropTypes from 'prop-types';
// Components
import WorkItem from 'components/WorkItem/WorkItem';
// Styles
import styles from './Timeline.scss';

const Timeline = (props) => {
  const { title, data } = props;
  return (
    <div className={styles.timeline}>
      <div className={styles.title}>{title}</div>
      {
        data.map((item) => (
          <div key={item.year} className={styles.event}>
            <div className={styles.year}>{item.year}</div>
            <div className={styles.content}>
              {item.type === 'work' && <WorkItem data={item.content}/>}
              {item.type === 'text' && item.content}
            </div>
          </div>
        ))
      }
    </div>
  );
};

Timeline.propTypes = {
  title: PropTypes.string,
  data: PropTypes.any,
};

export default Timeline;
