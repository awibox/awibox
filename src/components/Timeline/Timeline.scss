.timeline {
  margin-bottom: 50px;
  .timelineItem {
    position: relative;
    display: table;
    table-layout: fixed;
    width: 100%;
    padding-bottom: 15px;
    &:last-child {
      padding-bottom: 0;
      .right-part {
        padding-bottom: 0;
      }
    }
  }
  .leftPart {
    width: 20%;
    display: table-cell;
    padding-right: 25px;
    min-height: 100%;
    text-align: right;
    vertical-align: top;
  }
  .rightPart {
    width: 80%;
    display: table-cell;
    padding-left: 25px;
    padding-right: 15px;
    vertical-align: top;
    p, p > * {
      font-size: 0.92em;
    }
    ul {
      margin-left: 20px;
    }
  }
  .divider {
    position: absolute;
    top: 0;
    left: 20%;
    bottom: 0;
    width: 1px;
    background-color: #eee;
    &:before {
      content: '';
      display: block;
      position: absolute;
      margin-top: 4px;
      width: 17px;
      height: 17px;
      position: absolute;
      margin-left: -8px;
      border-radius: 10px;
      background-color: #fea621;
      opacity: .25;
      z-index: 0;
    }
    &:after {
      content: '';
      display: block;
      position: absolute;
      margin-top: 8px;
      width: 9px;
      height: 9px;
      margin-left: -4px;
      background-color: #fff;
      border-radius: 5px;
      border: 2px solid #fea621;
      z-index: 1;
    }
  }
  .itemTitle {
    font-size: 16px;
    margin-bottom: 3px;
  }
  .itemPeriod {
    margin: 3px 0;
    font-size: 14px;
    line-height: 1.4em;
  }
  .itemCompany {
    display: block;
    margin: 0 0 4px;
    font-size: 13px;
    font-weight: 300;
    line-height: 1.45em;
    color: #a5a6a7;
  }
  .item-logo {
    display: block;
    margin-bottom: 10px;
    img {
      max-height: 40px;
    }
  }
}

@media only screen and (max-width: 480px) {
  .timeline {
    .timelineItem {
      position: relative;
      display: block;
    }
    .leftPart {
      width: 100%;
      display: block;
      padding-left: 25px;
      min-height: 100%;
      text-align: left;
    }
    .divider {
      left: 0;
    }
    .rightPart {
      width: 100%;
      display: block;
      padding-left: 25px;
      padding-right: 0;
    }
  }
}
