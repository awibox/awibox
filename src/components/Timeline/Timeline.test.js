import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';
import Timeline from './Timeline';

const workData = List([
  {
    date: '2019 - Current',
    title: 'Lead Software Engineer',
    label: 'Com1',
    location: 'Saint Petersburg (Russia)',
    content: 'Text',
  },
  {
    date: '2010 - 2018',
    title: 'Lead Software Engineer',
    label: 'Com2',
    location: 'Saint Petersburg (Russia)',
    content: 'Text',
  },
]);

describe('Timeline', () => {
  it('should render correctly', () => {
    const component = shallow(<Timeline data={workData} />);
    expect(component).toMatchSnapshot();
  });
});
