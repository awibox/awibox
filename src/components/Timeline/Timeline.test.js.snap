// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline should render correctly 1`] = `
<div
  className="timeline clearfix"
>
  <div
    className="timelineItem clearfix"
    key="Com1"
  >
    <div
      className="leftPart"
    >
      <h5
        className="itemPeriod"
      >
        Com1
      </h5>
      <span
        className="itemCompany"
      >
        2019 - Current
      </span>
    </div>
    <div
      className="divider"
    />
    <div
      className="rightPart"
    >
      <h4
        className="itemTitle"
      >
        Lead Software Engineer
      </h4>
      Text
    </div>
  </div>
  <div
    className="timelineItem clearfix"
    key="Com2"
  >
    <div
      className="leftPart"
    >
      <h5
        className="itemPeriod"
      >
        Com2
      </h5>
      <span
        className="itemCompany"
      >
        2010 - 2018
      </span>
    </div>
    <div
      className="divider"
    />
    <div
      className="rightPart"
    >
      <h4
        className="itemTitle"
      >
        Lead Software Engineer
      </h4>
      Text
    </div>
  </div>
</div>
`;
